tests/functional/harris/harris_2d.py

```
@@ -141,28 +148,245 @@
+        self.simulator = None
+        self.plot_dir = None
+
+    def tearDown(self):
+        super(HarrisTest, self).tearDown()
+        if self.simulator is not None:
+            self.simulator.reset()
+        self.simulator = None
+        ph.global_vars.sim = None
+
+    def test_run(self):
+        self.register_diag_dir_for_cleanup(diag_dir)
```

Ruff (0.11.9)

Undefined name `diag_dir`

(F821)